framework/packages/abstract-core/src/native/ibc_host.rs

```diff
@@ -0,0 +1,167 @@
+
+    /// The BaseState contains the main addresses needed for sending and verifying messages
+    #[cosmwasm_schema::cw_serde]
+    pub struct Config {
+        /// AnsHost contract struct (address)
+        pub ans_host: AnsHost,
+        /// Address of the account factory, used to create remote accounts
+        pub account_factory: Addr,
+        /// Address of the local version control, for retrieving account information
+        pub version_control: VersionControlContract,
+    }
+
+    #[cosmwasm_schema::cw_serde]
+    pub struct ActionAfterCreationCache {
+        pub client_proxy_address: String,
+        pub account_id: AccountId,
+        pub action: HostAction,
+        pub chain_name: ChainName,
+    }
+}
+/// Used by Abstract to instantiate the contract
+/// The contract is then registered on the version control contract using [`crate::version_control::ExecuteMsg::ProposeModules`].
+#[cosmwasm_schema::cw_serde]
+pub struct InstantiateMsg {
+    /// Used to easily perform address translation on the app chain
+    pub ans_host_address: String,
+    /// Used to create remote abstract accounts
+    pub account_factory_address: String,
+    /// Version control address
+    pub version_control_address: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct MigrateMsg {}
+
+#[cosmwasm_schema::cw_serde]
+#[non_exhaustive]
+pub enum InternalAction {
+    /// Registers a new account from a remote chain
+    Register {
+        name: String,
+        description: Option<String>,
+        link: Option<String>,
+    },
+}
+
+#[cosmwasm_schema::cw_serde]
+#[non_exhaustive]
+pub enum HelperAction {
+    /// What do we need here ? TODO
+    SendAllBack,
+}
+
+/// Callable actions on a remote host
+#[cosmwasm_schema::cw_serde]
+pub enum HostAction {
+    Dispatch {
+        manager_msg: manager::ExecuteMsg,
+    },
+    /// Can't be called by an account directly. These are permissionned messages that only the IBC Client is allowed to call by itself.
+    Internal(InternalAction),
+    /// Some helpers that allow calling dispatch messages faster (for actions that are called regularly)
+    Helpers(HelperAction),
+}
+
+/// Interface to the Host.
+#[cosmwasm_schema::cw_serde]
+#[cfg_attr(feature = "interface", derive(cw_orch::ExecuteFns))]
+pub enum ExecuteMsg {
+    /// Update the Admin
+    UpdateAdmin { admin: String },
+    UpdateConfig {
+        ans_host_address: Option<String>,
+        account_factory_address: Option<String>,
+        version_control_address: Option<String>,
+    },
+    /// Register the Polytone proxy for a specific chain.
+    /// proxy should be a local address (will be validated)
+    RegisterChainProxy { chain: String, proxy: String },
+    /// Remove the Polytone proxy for a specific chain.
+    RemoveChainProxy { chain: String },
+    /// Create an account internally (used for account creation before account action)
+    InternalRegisterAccount {
```

Comment:
Can we move this into a `Callback` variant like with the other contracts?

Reply:
Or remove this alltogether by moving the reply into the account creation call (like mentioned in another comment)